colcon_cargo/task/cargo/__init__.py

```diff
@@ -1,30 +1,36 @@
 # Copyright 2018 Easymov Robotics
 # Licensed under the Apache License, Version 2.0
 
+from typing import Optional
+
 import os
 import shutil
 
 from colcon_core.environment_variable import EnvironmentVariable
 
-"""Environment variable to override the Cargo executable"""
+# Environment variable to override the Cargo executable
 CARGO_COMMAND_ENVIRONMENT_VARIABLE = EnvironmentVariable(
     'CARGO_COMMAND', 'The full path to the Cargo executable')
 
 
-def which_executable(environment_variable, executable_name):
+def which_executable(environment_variable: str, executable_name: str) \
+        -> Optional[str]:
     """
     Determine the path of an executable.
 
     An environment variable can be used to override the location instead of
     relying on searching the PATH.
-    :param str environment_variable: The name of the environment variable
-    :param str executable_name: The name of the executable
-    :rtype: str
+
+    :param environment_variable: The name of the environment variable
+    :param executable_name: The name of the executable, or None if that's not
+                            found
     """
-    value = os.getenv(environment_variable)
-    if value:
-        return value
-    return shutil.which(executable_name)
+    if environment_variable in os.environ:
```

Comment:
This change is equivalent to the existing code, why the change?

Reply:
I find  queyring directly the `os.environ` more readable in a couple of ways:

- There's no need for the intermediate `value` variable, which doesn't say much about the value it holds anyway
- the environment in python is just a dictionary, so using an `os` function, `getenv` , to manipulate it sounds unnecessary since there are already established ways of querying a dictionary.

In any case, it's not terribly important, I can revert the change if you think it was better before.